features/features.go

```diff
@@ -0,0 +1,96 @@
+//go:generate stringer -type=FeatureFlag
+
+package features
+
+import (
+	"expvar"
+	"fmt"
+	"sync"
+)
+
+type FeatureFlag int
+
+const (
+	unused                    FeatureFlag = iota
+	DNSAllowLoopbackAddresses             // Common
+	CheckMalformedCSR                     // WFE
+	DoNotForceCN                          // RA, CA
+	ReuseValidAuthz                       // RA
+	EnableMustStaple                      // CA
+	AcceptRevocationReason                // WFE
+)
+
+// List of features and their default value
+var features = map[FeatureFlag]bool{
+	unused: false,
+	DNSAllowLoopbackAddresses: false,
+	CheckMalformedCSR:         false,
+	DoNotForceCN:              false,
+	ReuseValidAuthz:           false,
+	EnableMustStaple:          false,
+	AcceptRevocationReason:    false,
+}
+
+var fMu = new(sync.RWMutex)
+
+var initial = map[FeatureFlag]bool{}
+
+var nameToFeature = make(map[string]FeatureFlag, len(features))
+
+func init() {
+	for f, v := range features {
+		nameToFeature[f.String()] = f
+		initial[f] = v
+	}
+}
+
+type boolVar bool
+
+// you'd think bool would implement this itself
+func (b boolVar) String() string { return fmt.Sprintf("%t", b) }
+
+// Set accepts a list of features and whether they should
+// be enabled or disabled, it will return a error if passed
+// a feature name that it doesn't know
+func Set(featureSet map[string]bool) error {
+	fMu.Lock()
+	defer fMu.Unlock()
+	for n, v := range featureSet {
+		f, present := nameToFeature[n]
+		if !present {
+			return fmt.Errorf("feature '%s' doesn't exist", n)
```

Comment:
I like that we're checking for misspellings. Just to keep in mind: This means that deprecating a flag will consist of moving it to a "deprecated" section of the features list, then removing the flag in prod, then deleting it from the features list.